File: 투포인터/주몽.java
package 투포인터;

import java.util.Arrays;
import java.util.Scanner;

public class 주몽 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // 입력받기
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        int[] arrays = new int[n];

        for (int i = 0; i < n; i++) {
            arrays[i] = scanner.nextInt();
        }

        // 배열 정렬
        Arrays.sort(arrays);

        int answer = 0;
        int start = 0;
        int end = n - 1;

        // 투 포인터 알고리즘
        while (start < end) {
            int currentValue = arrays[start] + arrays[end];

            if (currentValue == m) {
                start++;
                end--;
                answer++;
            } else if (currentValue > m) {
                end--;
            } else { // currentValue < m
                start++;
            }
        }

        System.out.println(answer);
    }
}

File: 투포인터/주몽.py
n = int(input())
m = int(input())

arrays = list(map(int, input().split()))

arrays.sort()
answer = 0

s = 0
e = n - 1

while(True):
    if s >= e:
        break

    current_value = arrays[s] + arrays[e]

    if(current_value == m):
        s += 1
        e -= 1
        answer += 1
    elif (current_value > m):
        e -= 1
    elif (current_value < m):
        s += 1

print(answer)